### PR TITLE
Handle if Rojo server ends due to crash

### LINKED ISSUE #31: Update the status icon if Rojo crashes
Currently, if Rojo crashes, the status icon still remains as if it is currently running. This can be confusing when you are unsure why files will no longer sync, or why the studio plugin will not connect. You have to explicitly check the Rojo extension output to see if that is the issue.

It would make it easier if the icon would update if Rojo is no longer running, either back to the "Start Rojo" status, or a status showing explicitly that Rojo has stopped and it wasn't because the stop command was run.

### PR DESCRIPTION
**Pull request by JohnnyMorganz**
Small UX update to handle when Rojo unexpectedly quits due to an error.
Currently, if Rojo crashes, there is no indication of such - this can be confusing (especially for newer developers), as the button still indicates that a server is running, and you do not realise that the server has stopped until you explicitly open up the output of the extension and check.

This PR implements a new state to the button: "Crashed". If Rojo exits with a non-zero exit code, then we set the button state to "crashed". We also automatically open up the Rojo console output, so that the end user can quickly see the error that occured.

Closes #31

### CODE REVIEW

File: src/Rojo.ts [hunk 1]
```diff
@@ -224,6 +224,18 @@ export class Rojo<C extends object = {}> extends vscode.Disposable {
       })
     )
 
+    this.server.on(
+      "exit",
+      // Code can possibly be "null", for example if the SIGTERM signal is sent
+      (code: number | null, _signal) => {
+        // Check if exited with a non-zero exit code (i.e. an error occured)
+        if (code && code !== 0) {
+          this.outputChannel.show()
```
Comment (evaera) [inline src/Rojo.ts:233]: It looks like to me that the code already shows the output channel when anything is printed to stderr:

https://github.com/rojo-rbx/vscode-rojo/blob/4940ab4be3811b330f3c1b88e1e0b171c60493a1/src/Rojo.ts#L216

So this should be an indication that it crashed. Is this not showing for you? It does for me, when I force a crash in my testing.

**Review by evaera — COMMENTED** [review]
I tested this with multiple project files serving at the same time and it seems to work as expected!

File: src/Rojo.ts (hunk 1 above)
Comment (JohnnyMorganz) [reply src/Rojo.ts:233]: Hm, I may well have missed this somehow - odd. I can't remember exactly, but I'll get back to you

Comment (JohnnyMorganz) [reply src/Rojo.ts:233]: You're right, it does show the output channel when crashed! I'm confused on how I missed that, I probably wasn't paying attention properly.

I guess this PR does add a small QoL update re: the button, but it seems even less necessary now - I understand if its too minor, feel free to close if you believe so

**Pull request merged**